### PR TITLE
[docs-only] Add documentation for proxy routes

### PR DESCRIPTION
**Pull request by kobergj**
Fixes: #7749 (Proxy service readme update needed)

Documents proxy route configuration

### CODE REVIEW

File: services/proxy/README.md [hunk 1]
```diff
@@ -13,6 +13,35 @@ The following request authentication schemes are implemented:
 -   Signed URL
 -   Public Share Token
 
+## Configuring routes
+
+The proxy handles routing to all endpoints that ocis offers. The current default routes can be found here https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go
+Due to complexity of the configuration, new routes can only be defined in the yaml file.
+For overwriting the default routes use:
```
Comment (mmattel) [inline services/proxy/README.md:21]:
```suggestion
Due to the complexity when defining routes, these can only be defined in the yaml file but not via environment variables.

For _overwriting_ default routes, use the following yaml example:

```

File: services/proxy/README.md [hunk 2]
```diff
@@ -13,6 +13,35 @@ The following request authentication schemes are implemented:
...
+```
+For adding additional routes to the default routes use:
+```yaml
+additional_policies:
+  - name: ocis
+    routes:
+      - endpoint: /custom/endpoint
+        service: com.owncloud.cutom.custom
+```
+A route has the following configurable parameters:
```
Comment (mmattel) [inline services/proxy/README.md:39]:
```suggestion

A route has the following configurable parameters:

```

File: services/proxy/README.md [hunk 3]
```diff
@@ -13,6 +13,35 @@ The following request authentication schemes are implemented:
...
+```yaml
+endpoint: "" # the url that should be routed
+service: "" # the service the url should be routed to
+unprotected: false # skip authorization
```
Comment (mmattel) [inline services/proxy/README.md:43]:
```suggestion
endpoint: ""       # the url that should be routed
service: ""        # the service the url should be routed to
unprotected: false # skip authorization
```
Renders hashes underneath making the keys better readble
What happens with `unprotected` true?

Comment (kobergj) [reply services/proxy/README.md:43]: `unprotected: true` => No authorization checks. This is an open endpoint
`unprotected: false` => Authorization is checked on request, user needs an auth header to access this endpoint

File: services/proxy/README.md [hunk 4]
```diff
@@ -13,6 +13,42 @@ The following request authentication schemes are implemented:
...
+```
+
+For adding _additional_ routes to the default routes use:
+
+```yaml
+additional_policies:
+  - name: ocis
+    routes:
+      - endpoint: /custom/endpoint
+        service: com.owncloud.cutom.custom
```
Comment (mmattel) [inline services/proxy/README.md:41]:
```suggestion
        service: com.owncloud.custom.custom
```
typo ? 🤣

File: services/proxy/README.md [hunk 5]
```diff
@@ -13,6 +13,42 @@ The following request authentication schemes are implemented:
 -   Signed URL
 -   Public Share Token
 
+## Configuring Routes
+
+The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go).
```
Comment (mmattel) [inline services/proxy/README.md:18]:
```suggestion
The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary, when writing own ocis extensions.
```

File: services/proxy/README.md [hunk 6]
```diff
@@ -13,6 +13,42 @@ The following request authentication schemes are implemented:
 -   Signed URL
 -   Public Share Token
 
+## Configuring Routes
+
+The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary, when writing own ocis extensions.
```
Comment (mmattel) [inline services/proxy/README.md:18]:
```suggestion
The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary when writing own ocis extensions.
```

**Review by mmattel — APPROVED** [review]
LGTM 👍

**Pull request merged**